thingsboard_gateway/connectors/ftp/path.py

```diff
@@ -0,0 +1,140 @@
+import os
+
+from thingsboard_gateway.connectors.connector import log
+from thingsboard_gateway.connectors.ftp.file import File
+
+COMPATIBLE_FILE_EXTENSIONS = ('json', 'txt', 'csv')
+
+
+class Path:
+    def __init__(self, path: str, with_sorting_files: bool, poll_period: int, read_mode: str, max_size: int,
+                 delimiter: str, telemetry: list, device_name: str, device_type: str, attributes: list,
+                 txt_file_data_view: str):
+        self._path = path
+        self._with_sorting_files = with_sorting_files
+        self._poll_period = poll_period
+        self._files: [File] = []
+        self._delimiter = delimiter
+        self._last_polled_time = 0
+        self._telemetry = telemetry
+        self._attributes = attributes
+        self._device_name = device_name
+        self._device_type = device_type
+        self._txt_file_data_view = txt_file_data_view
+        self.__read_mode = File.ReadMode[read_mode]
+        self.__max_size = max_size
+
+    @staticmethod
+    def __is_file(ftp, filename):
+        current = ftp.pwd()
+        try:
+            ftp.cwd(filename)
+        except Exception:
+            ftp.cwd(current)
+            return True
+        ftp.cwd(current)
+        return False
+
+    @staticmethod
+    def __folder_exist(ftp, folder_name):
+        current = ftp.pwd()
+
+        try:
+            ftp.cwd(folder_name)
+        except Exception:
+            ftp.cwd(current)
+            return False
+        ftp.cwd(current)
+        return True
+
+    def __get_files(self, ftp, paths, file_name, file_ext):
+        kwargs = {}
+        for item in paths:
+            ftp.cwd(item)
+
+            folder_and_files = ftp.nlst()
+
+            for ff in folder_and_files:
+                cur_file_name, cur_file_ext = ff.split('.')
+                if cur_file_ext in COMPATIBLE_FILE_EXTENSIONS and self.__is_file(ftp, ff):
```

Comment:
What if we will have a filename like "*_10082021.log"? Could we use a regex search with partial flag here?